shared/common-adapters/tab-bar.js.flow

```diff
@@ -31,9 +30,11 @@ export type ItemProps = Exact<{
 
 export class TabBarItem extends Component<void, ItemProps, void> { }
 
+export type TabBarButtonSource = {type: 'icon', icon: IconType} | {type: 'avatar', avatar: Avatar} | {type: 'nav', icon: IconType}
+
 export type TabBarButtonProps = Exact<{
-  selected: boolean,
-  source: {type: 'icon', icon: IconType} | {type: 'avatar', avatar: Avatar} | {type: 'nav', icon: IconType},
+  selected?: boolean,
+  source?: TabBarButtonSource,
```

Comment:
I don't think it makes sense to make this optional. Maybe you should add another variant to what you are trying to do. Which seems like just a label?